third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -171,7 +172,7 @@ impl<'r> TransactionDataCache<'r> {
         Ok(change_set)
     }
 
-    pub(crate) fn num_mutated_accounts(&self, sender: &AccountAddress) -> u64 {
+    pub(crate) fn num_mutated_resources(&self, sender: &AccountAddress) -> u64 {
```

Comment:
Yes, because it uses `data_map` in its implementation and not modules. This nicely aligns with the fact that we will not store modules inside anymore👍

To be honest, I am not sure how useful this API is, it is only used for tests, but I know that not everyone is happy with removing public interfaces no matter what😄 But this seems so ugly... I agree, why we assume sender is mutated? Mu guess is that this must be the adapter's responsibility, where we have senders.

Example of a test:
```
// The resource was published to "account1" and the sender's account
// (TEST_ADDR) is assumed to be mutated as well (e.g., in a subsequent
// transaction epilogue).
assert_eq!(sess.num_mutated_resources(&TEST_ADDR), 2);
```
But this is just wrong, test assumes that something is mutated....

Reply:
So I see two ways:
- Remove this assumption on sender. Move doesn't care about this and epilogue, this is clearly off.
- Remove this API completely, or better make it `#[cfg(feature = "testing")]` or something like that.
@gelash what do you think?

Reply:
cc @runtian-zhou @vgao1996 who might have some context about this

Reply:
Ping @runtian-zhou @vgao1996